### DISCUSSION TITLE
How can I 'un-active' header if certain criteria meets?

### DISCUSSION THREAD

**Discussion by TetraTheta** [hbstack/.github#44] [Q&A] [answered]
I have header menu defined in `menu.yml` to be like this:
> * Chit Chat
> * Game
>     * Some
>     * Other
>     * Games
> * Genshin Impact
>     * Some
>     * Other
>     * Stuffs
>     * Misc

But, actual file structure is like this:
```plaintext
content/blog
├ chit-chat
└ game
  ├ some
  ├ other
  ├ games
  └ genshin-impact
    ├ some
    ├ other
    ├ stuffs
    └ misc
```
This menu and actual file structure unmatch is intended.  

Many Hugo theme including HB Theme (Sorry, I forgot what theme I'm currently using is now called as) doesn't support nested child menu like [Game - Genshin Impact - Misc].  

This is only way to show child menu of 'Genshin Impact' to header, by defining 'Genshin Impact' child menu as parent menu in `menu.yml`.

But in this layout, both 'Game' and 'Genshin Impact' header menu is set as `active` for posts in 'Genshin Impact' because [`is-current.html` function](https://github.com/hbstack/header/blob/main/layouts/partials/hb/modules/header/functions/is-current.html) will return `true` for both 'Game' and 'Genshin Impact' menu, thus highlighting them.

![1](https://github.com/hbstack/.github/assets/37061801/b1e29c01-6935-4688-a933-4fd76c31a6f1)

I want to 'un-active' 'Game' menu if 'Genshin Impact' menu is also active.

![2](https://github.com/hbstack/.github/assets/37061801/1b6f93de-796b-4049-b986-d811a4006dea)

But I don't know what file to modify for this. [`is-current.html`](https://github.com/hbstack/header/blob/main/layouts/partials/hb/modules/header/functions/is-current.html)? [`menu.html`](https://github.com/hbstack/header/blob/main/layouts/partials/hb/modules/header/menu.html)? Also, how should I modify it?

Sorry if this question looks like asking for spoon-feeding. I've completely lost my way of how to modify things to achieve my goal.

**Comment by razonyang** [comment, marked as answer]
Create file `layouts/partials/hb/modules/header/functions/is-current.html` on your site with the same content, and remove line `3-12`.

> Any files with same path from your site root will get higher priority and override the theme/module's, but that also mean, you may not get updated fully when upstream theme/module changed.

https://github.com/hbstack/header/blob/e2c6b84705526ba39fbe7ba16b057672b59a372c/layouts/partials/hb/modules/header/functions/is-current.html#L3-L12

I haven't test, tweak it to suit your needs.

**Reply by TetraTheta** [reply]
Thank you! I've managed to tweak it to my goal.
```
{{/* Indicate whether the menu is the current page. */}}
{{- $v := or (.Page.IsMenuCurrent "main" .Menu) (.Page.HasMenuCurrent "main" .Menu) -}}
{{- if ne .Menu.URL "" -}}
  {{- $homeURL := strings.TrimSuffix "/" site.Home.Permalink -}}
  {{- $menuURL := strings.TrimSuffix "/" (absURL .Menu.URL) -}}
  {{- $pageURL := strings.TrimSuffix "/" .Page.Permalink -}}
  {{- $v = or $v (eq $menuURL $pageURL) (and (ne $menuURL $homeURL) (hasPrefix $pageURL $menuURL)) -}}
  {{/* Filter Genshin Impact */}}
  {{- with and (strings.Contains $menuURL "game") (not (strings.Contains $menuURL "genshin-impact")) (strings.Contains $pageURL "genshin-impact") -}}
    {{- $v = not . -}}
  {{- end -}}
{{- end -}}
{{- return $v -}}
```
I'm not sure about its efficiency, but at least, it works as I want.
